tests/frontend/specs/timeslider_labels.js

```diff
@@ -2,63 +2,61 @@ describe("timeslider", function(){
-    this.timeout(60000);
-    for(var i=0; i < revs; i++) {
-      setTimeout(function() {
-        // enter 'a' in the first text element
-        inner$("div").first().sendkeys('a');
-      }, 200);
+  /**
+   * @todo test authorsList
+   */
+  it("Shows a date and time in the timeslider and make sure it doesn't include NaN", async function() {
+    // make some changes to produce 3 revisions
+    let revs = 3;
+
+    for(let i=0; i < revs; i++) {
+      await helper.edit('a\n');
     }
 
-    setTimeout(function() {
-      // go to timeslider
-      $('#iframe-container iframe').attr('src', $('#iframe-container iframe').attr('src')+'/timeslider');
-
-      setTimeout(function() {
-        var timeslider$ = $('#iframe-container iframe')[0].contentWindow.$;
-        var $sliderBar = timeslider$('#ui-slider-bar');
-
-        var latestContents = timeslider$('#padcontent').text();
-
-        // Expect the date and time to be shown
-
-        // Click somewhere on the timeslider
-        var e = new jQuery.Event('mousedown');
-        e.clientX = e.pageX = 150;
-        e.clientY = e.pageY = 45;
-        $sliderBar.trigger(e);
-
-        e = new jQuery.Event('mousedown');
-        e.clientX = e.pageX = 150;
-        e.clientY = e.pageY = 40;
-        $sliderBar.trigger(e);
-
-        e = new jQuery.Event('mousedown');
-        e.clientX = e.pageX = 150;
-        e.clientY = e.pageY = 50;
-        $sliderBar.trigger(e);
-
-        $sliderBar.trigger('mouseup')
-
-        setTimeout(function() {
-          //make sure the text has changed
-          expect( timeslider$('#timer').text() ).not.to.eql( "" );
-          expect( timeslider$('#revision_date').text() ).not.to.eql( "" );
-          expect( timeslider$('#revision_label').text() ).not.to.eql( "" );
-          var includesNaN = timeslider$('#revision_label').text().indexOf("NaN"); // NaN is bad. Naan ist gut
-          expect( includesNaN ).to.eql( -1 ); // not quite so tasty, I like curry.
-          done();
-        }, 400);
-      }, 2000);
-    }, 2000);
+    await helper.gotoTimeslider(revs);
+    await helper.waitForPromise(function(){return helper.contentWindow().location.hash === '#'+revs})
+
+    // the datetime of last edit
+    let timerTimeLast = new Date(helper.timesliderTimerTime()).getTime();
+
+    // the day of this revision, e.g. August 12, 2020 (stripped the string "Saved")
+    let dateLast = new Date(helper.revisionDateElem().substr(6)).getTime();
+
+    // the label/revision, ie Version 3
+    let labelLast = helper.revisionLabelElem().text();
+
+    // the datetime should be a date
+    expect( Number.isNaN(timerTimeLast)).to.eql(false);
+
+    // the Date object of the day should not be NaN
+    expect( Number.isNaN(dateLast) ).to.eql(false)
+
+    // the label should match Version `Number`
+    expect( labelLast.indexOf(`Version ${revs}`) ).to.not.be(-1);
+
+    // Click somewhere left on the timeslider to go to revision 0
+    helper.sliderClick(30);
```

Comment:
Because there are three revisions, I would expect that clicking at 30 would go to revision 1. To be safe, maybe click more to the left:

```suggestion
    helper.sliderClick(1);
```

Reply:
afaik I tested it and it turned out to be not so safe as it should be (maybe the calculations are wrong and it actually clicks outside?) will test again